modules/quadbin/snowflake/sql/QUADBIN_SIBLING.sql

```diff
@@ -0,0 +1,21 @@
+----------------------------
+-- Copyright (C) 2022 CARTO
+----------------------------
+
+CREATE OR REPLACE FUNCTION QUADBIN_SIBLING
+(quadbin INT, direction STRING)
+RETURNS INT
+AS $$
+    CASE
+        WHEN direction = 'left' THEN
+            _QUADBIN_SIBLING(quadbin, -1, 0)
+        WHEN direction = 'right' THEN
+            _QUADBIN_SIBLING(quadbin, 1, 0)
+        WHEN direction = 'up' THEN
+            _QUADBIN_SIBLING(quadbin, 0, -1)
+        WHEN direction = 'down' THEN
+            _QUADBIN_SIBLING(quadbin, 0, 1)
+        ELSE 
+            NULL
+    END
+$$;
```

Comment:
You could integrate the auxiliar functions into the main one as:

```sql
CREATE OR REPLACE FUNCTION QUADBIN_SIBLING
(quadbin INT, direction STRING)
RETURNS INT
AS $$
  (WITH _dxy AS (
    SELECT CASE direction
        WHEN 'left' THEN
            OBJECT_CONSTRUCT('x', -1, 'y', 0)
        WHEN 'right' THEN
            OBJECT_CONSTRUCT('x', 1, 'y', 0)
        WHEN 'up' THEN
            OBJECT_CONSTRUCT('x', 0, 'y', -1)
        WHEN 'down' THEN
            OBJECT_CONSTRUCT('x', 0, 'y', 1)
        ELSE 
            NULL
    END AS d, QUADBIN_TOZXY(quadbin) AS t    
  )
  SELECT IFF(t:y + d:y >= 0 AND t:y + d:y < BITSHIFTLEFT(1, t:z),
        QUADBIN_FROMZXY(
            t:z,
            MOD(t:x + d:x, BITSHIFTLEFT(1, t:z)) + IFF(t:x + d:x < 0, BITSHIFTLEFT(1, t:z), 0),
            t:y + d:y
        ),
        NULL
    )
   FROM _dxy)
$$;
```

I wrote that function with the hope to avoid the dreaded crashes when using it on a table but it still crashes... I'm afraid we'd need to inline the calls to QUADBIN_TOZXY and QUADBIN_FROMZXY as well, which is nuts :(